File: DotNet/CesiumLanguageWriter/Advanced/CesiumDistanceDisplayConditionValuePropertyAdaptor.cs
using JetBrains.Annotations;

namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// Adapts a class that implements <see cref="ICesiumDistanceDisplayConditionValuePropertyWriter"/> to implement
    /// <see cref="ICesiumInterpolatableValuePropertyWriter{TValue}"/> for <see cref="Bounds"/> values.
    /// </summary>
    /// <typeparam name="TFrom">The class that implements <see cref="ICesiumDistanceDisplayConditionValuePropertyWriter"/> to adapt.</typeparam>
    public class CesiumDistanceDisplayConditionValuePropertyAdaptor<TFrom> : CesiumInterpolatableWriterAdaptor<TFrom, Bounds>
        where TFrom : class, ICesiumDistanceDisplayConditionValuePropertyWriter
    {
        /// <summary>
        /// Initializes a new instance.
        /// </summary>
        /// <param name="parent">The instance to wrap.</param>
        /// <param name="writeValueCallback">The callback to write values of type <see cref="Bounds"/>.</param>
        /// <param name="writeSamplesCallback">The callback to write samples of type <see cref="Bounds"/>.</param>
        /// <param name="writeDeleteValueCallback">The callback to write an indication that the client should delete existing data.</param>
        public CesiumDistanceDisplayConditionValuePropertyAdaptor([NotNull] TFrom parent,
                                                                  [NotNull] CesiumWriterAdaptorWriteCallback<TFrom, Bounds> writeValueCallback,
                                                                  [NotNull] CesiumWriterAdaptorWriteSamplesCallback<TFrom, Bounds> writeSamplesCallback,
                                                                  [NotNull] CesiumWriterAdaptorWriteDeleteCallback<TFrom> writeDeleteValueCallback)
            : base(parent, writeValueCallback, writeSamplesCallback, writeDeleteValueCallback)
        {
        }
    }
}

File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/CesiumDistanceDisplayConditionValuePropertyAdaptor.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;
import javax.annotation.Nonnull;

/**
 * Adapts a class that implements {@link ICesiumDistanceDisplayConditionValuePropertyWriter} to implement
 {@link ICesiumInterpolatableValuePropertyWriter} for {@link Bounds} values.
 * @param <TFrom> The class that implements {@link ICesiumDistanceDisplayConditionValuePropertyWriter} to adapt.
 */
@SuppressWarnings( {
        "unused",
        "deprecation",
        "serial"
})
public class CesiumDistanceDisplayConditionValuePropertyAdaptor<TFrom extends ICesiumDistanceDisplayConditionValuePropertyWriter> extends CesiumInterpolatableWriterAdaptor<TFrom, Bounds> {
    /**
    * Initializes a new instance.
    * @param parent The instance to wrap.
    * @param writeValueCallback The callback to write values of type {@link Bounds}.
    * @param writeSamplesCallback The callback to write samples of type {@link Bounds}.
    * @param writeDeleteValueCallback The callback to write an indication that the client should delete existing data.
    */
    public CesiumDistanceDisplayConditionValuePropertyAdaptor(@Nonnull TFrom parent, @Nonnull CesiumWriterAdaptorWriteCallback<TFrom, Bounds> writeValueCallback,
            @Nonnull CesiumWriterAdaptorWriteSamplesCallback<TFrom, Bounds> writeSamplesCallback, @Nonnull CesiumWriterAdaptorWriteDeleteCallback<TFrom> writeDeleteValueCallback) {
        super(parent, writeValueCallback, writeSamplesCallback, writeDeleteValueCallback);
    }
}